File: frontend/src/types/session.ts
export interface SessionSummary {
  id: string;
  agent_id: string;
  title: string;
  created_at: string;
  updated_at: string;
  preview: string;
  mode: string | null;
}

export interface SessionListResponse {
  sessions: SessionSummary[];
}

export interface ChipInfo {
  id: string;
  label: string;
  enabled: boolean;
  disabled: boolean;
}

export interface ContentModule {
  id: string;
  title: string;
  chips: ChipInfo[];
  content: Record<string, unknown>;
}

export interface SessionDetail {
  id: string;
  agent_id: string;
  title: string;
  created_at: string;
  updated_at: string;
  preview: string;
  mode: string | null;
  modules: ContentModule[];
  messages: ChatMessage[];
}

export interface ChatMessage {
  id?: string;
  role: "user" | "assistant";
  content: string;
  timestamp?: string;
}

export interface PinnedItem {
  id: string;
  module_id: string;
  chip_id: string;
  title: string;
  content: string;
}

export interface Report {
  session_id: string;
  title: string;
  markdown: string;
  updated_at: string;
}

export interface ScheduleConfig {
  frequency: string;
  day_of_week: string | null;
  time: string;
  minutes_before: number | null;
  recipients: string;
}

File: backend/app/schemas/session.py
from typing import Any, Optional
from pydantic import BaseModel


class SessionSummary(BaseModel):
    id: str
    agent_id: str
    title: str
    created_at: str
    updated_at: str
    preview: str
    mode: Optional[str] = None
    subject: Optional[str] = None


class SessionListResponse(BaseModel):
    sessions: list[SessionSummary]


class SessionCreateRequest(BaseModel):
    agent_id: str
    title: Optional[str] = None
    subject: Optional[str] = None


class ChipInfo(BaseModel):
    id: str
    label: str
    enabled: bool


class ContentModule(BaseModel):
    id: str
    title: str
    chips: list[ChipInfo]
    content: dict[str, Any]


class ChatMessageSchema(BaseModel):
    id: str
    role: str
    content: str
    timestamp: Optional[str] = None


class SessionDetail(BaseModel):
    id: str
    agent_id: str
    title: str
    created_at: str
    updated_at: str
    preview: str
    mode: Optional[str] = None
    modules: list[ContentModule]
    messages: list[ChatMessageSchema]


class ApplyChatRequest(BaseModel):
    content: str
